src/gui/ui/opengl_setup.cpp

```diff
@@ -0,0 +1,80 @@
+#include "opengl_setup.hpp"
+
+#include "../../logging.hpp"
+
+namespace gui {
+
+GLFWwindow*
+setup_opengl(screen_size_t window_width, screen_size_t window_height) {
+    glEnable(GL_MULTISAMPLE);
+
+    // initialize logging
+    glfwWindowHint(GLFW_OPENGL_DEBUG_CONTEXT, true);
+
+    // Initialise GLFW
+    glewExperimental = true; // Needed for core profile
+    if (!glfwInit()) {
+        LOG_CRITICAL(logging::opengl_logger, "Failed to initialize GLFW");
+        getchar();
+        glfwTerminate();
+        return nullptr;
```

Comment:
std optional pls

Reply:
this returns a window pointer, not sure if one can instantiate an optional from a pointer.

Reply:
Optional can hold anything

It was created to hold ptrs, so they would never be null.

Reply:
>Thus, an optional object models an object, not a pointer, even though operator*() and operator->() are defined. 

https://en.cppreference.com/w/cpp/utility/optional

GLFWwindow is a strut, but I can't really know anything about the implimentation. It's never dereferenced, and passed like its constant.

Reply:
Its opaque. Perfect use of std optional.